### PR TITLE
[POAE7-2382] add cider feature user-guide doc and reorg dev-guide doc

### PR DESCRIPTION
**Pull request by kd97**
### What changes were proposed in this pull request?

add cider feature user-guide doc and reorg dev-guide doc
### Why are the changes needed?

reorg cider doc.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

local test

### CODE REVIEW

File: docs/user/data-type.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+Data Types
```
Comment (yma11) [inline docs/user/data-type.rst:1]: have 2 comments here:
1) you may refer some information from [here](https://docs.heavy.ai/sql/data-definition-ddl/datatypes-and-fixed-encoding#datatypes).
2) maybe we should don't put these detailed introduction as user manual? For user, what they need is more about how to build, install, configuration guide, trigger and may some easy debug guide.

File: docs/user/function-support.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
+Function support
+=====================
+
+Aggregate Functions
+--------------------------------------
+**avg(x)**
```
Comment (yma11) [inline docs/user/function-support.rst:6]: For information like this, which is almost common info for SQL, we may can put them as reference? like https://openmldb.ai/docs/zh/main/reference/index.html.

File: docs/develop/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
+======================
+Contributing
+======================
+
+Code Style Check
+------------------------
+Code style check will be triggered automatically after you submit a PR. So please ensure your PR does not break any of these workflows. It runs ``make format-check``, ``make header-check`` as part of our continuous integration. 
+Pull requests should pass ``format-check`` and ``header-check`` without errors before being accepted.
+
+More details can be found at `ci/scripts/run_cpplint.py <https://github.com/intel-innersource/frameworks.ai.modular-sql.velox-plugin/blob/40591b915bfee8068749218725f9c95a4704bacd/ci/scripts/run_cpplint.py>`_
```
Comment (jikunshang) [inline docs/develop/contributing.rst:10]: link

**Review by jikunshang — COMMENTED** [review]
Anyway to have a preview in web page before merge?

**Comment by kd97** [comment]
> Anyway to have a preview in web page before merge?

Now need to run sphinx locally on gh-pages branch and check for rst file format issues.

**Review by yma11 — DISMISSED** [review]

**Comment by jikunshang** [comment]
Please rebase code to latest.

**Review by jikunshang — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Pull request merged**